src/matcher/equals.hpp

```diff
@@ -64,4 +68,32 @@ template <typename T> class equals : public base_impl<equals<T>> {
     friend class base_impl<equals<T>>;
 };
 
+template <> class equals<double> : public base_impl<equals<double>> {
+public:
+    using rule_data_type = std::vector<std::pair<std::string_view, uint64_t>>;
+
+    // NOLINTNEXTLINE(bugprone-easily-swappable-parameters)
+    equals(double expected, double precision) : expected_(expected), precision_(precision) {}
```

Comment:
I'd call it _delta_ or _dx_, because the usual meaning "precision"  when dealing with quantities is the effective number of digits in the numbers. What you have here is the "absolute uncertainty".

Reply:
You're right, precision in this context is not the right name for it, I'll rename it to delta.